src/components/Questions/Palette/QuestionPalette.vue

```diff
@@ -18,20 +18,24 @@
     </div>
 
     <div
-      class="grid grid-cols-5 bp-500:grid-cols-6 lg:grid-cols-7 xl:grid-cols-8 mt-4 space-y-4"
-    >
-      <PaletteItem
-        v-for="(itemState, index) in questionStates"
-        class="hover:cursor-pointer"
-        :class="{ 'mt-4': index == 0 }"
-        :key="index"
-        :index="index"
-        :hasQuizEnded="hasQuizEnded"
-        :state="itemState.value"
-        :isHighlighted="currentQuestionIndex == itemState.index"
-        @click="navigateToQuestion(itemState.index)"
-        :data-test="`paletteItem-${itemState.index}`"
-      ></PaletteItem>
+      v-for="(itemSetState, index) in questionSetStates" :key="index" class="space-y-2">
+        <p :class="titleTextClass" :data-test="`paletteTitle-${index}`">{{ itemSetState.title }}</p>
+        <p :class="instructionTextClass" :data-test="`paletteInstruction-${index}`">{{ itemSetState.instructionText }}</p>
+        <div class="grid grid-cols-5 bp-500:grid-cols-6 lg:grid-cols-7 xl:grid-cols-8 mt-4 space-y-4">
+          <PaletteItem
+            v-for="(questionItemState, qindex) in itemSetState.paletteItems"
+            class="hover:cursor-pointer"
+            :class="{ 'mt-4': qindex == 0 }"
+            :key="qindex"
+            :index="questionItemState.index"
+            :hasQuizEnded="hasQuizEnded"
+            :state="questionItemState.value"
+            :isHighlighted="currentQuestionIndex == questionItemState.index"
+            @click="navigateToQuestion(questionItemState.index)"
+            :data-test="`paletteItem-${questionItemState.index}`"
+          ></PaletteItem>
+        </div>
+
```

Comment:
we should decide on this and keep only one thing in the code, either question/questionSet or item/itemSet